agent/www/components/addresses/addresses_ctrl.js

```diff
@@ -214,26 +214,33 @@ angular.module('patternfly.toolbars').controller('ViewCtrl', ['$scope', '$timeou
           }
         };
 
-        function is_valid_filter(filter) {
-            if (filter.title === 'Name') {
-                try {
-                    $scope.notification.show_alert = false;
+        var filterChange = function (filters) {
+            $scope.filtersText = "";
 
-                    var dummy = new RegExp(filter.value);
+            $scope.notification.show_alert = false;
+            var messages = [];
+            var valid = filters.filter(function (filter) {
```

Comment:
Not a big issue, but was there a reason to inline the is_valid_filter function? (To me the flow is more readable as a separate, named function).

Reply:
@vbusch's previous change had changed `is_valid_filter` from a pure function to be one with a side effect (namely the manipulation of the notification).   For correct behaviour the 'bad' filter always needed to appear on the end of the list, otherwise the next good filter would have reset the notification set.   Currently, as new filters are pushed to the end of the list, this was a safe assumption, but it felt fragile to me: it was easy to imagine a innocuous change elsewhere breaking this code.

Inlining the function allowed me to push the responsibility for manipulating notification state to filterChange whilst still calling new RegExp() only once per filter.  I can refactor again to to reinstate the pure function and separately build up the validation messages if you prefer.

Reply:
No, thats a good explanation!